### ISSUE TITLE
Frontmatter in daily template overwritten

### ISSUE THREAD

**Issue by gbbirkisson** [epwalsh/obsidian.nvim#494]
### 🐛 Describe the bug

I have this template configured for daily notes:

```markdown
---
id: "{{today}}"
aliases:
  - Daily {{today}}
  - "{{today}}"
tags: []
day: {{day}}
month: {{month}}
week: {{week}}
weekday: {{weekday}}
weekdayname: {{weekdayname}}
year: {{year}}
---
# {{today}}

What is going on?
```

When running `:ObsidianToday`, and a new file is created, I get this output:

```markdown
---
id: "2024-03-15"
aliases:
  - mars 15, 2024
tags:
  - daily-notes
---

# 2024-03-15

What is going on?
```

As you can see the frontmatter is gone! This used to work, I don't know why it stopped working...

If I run `:ObsidianTemplate daily.md` I get the output that I expect:

```markdown
---
id: "2024-03-15"
aliases:
  - Daily 2024-03-15
  - "2024-03-15"
tags: []
day: 15
month: 03
week: 11
weekday: 6
weekdayname: Friday
year: 2024
---

# 2024-03-15

What is going on?
```

### Config

```lua
  {
    'epwalsh/obsidian.nvim',
    version = 'v3.7.3',
    lazy = true,
    cmd = { 'ObsidianToday', 'ObsidianCheck' },
    dependencies = {
      'nvim-lua/plenary.nvim',
    },
    opts = {
      workspaces = {
        {
          name = 'personal',
          path = '~/repos/personal/notes',
        },
      },
      daily_notes = {
        folder = '0-daily',
        template = 'daily.md',
      },
      templates = {
        subdir = '6-extras/templates',
        date_format = '%Y-%m-%d',
        time_format = '%H:%M',
        substitutions = {
          yesterday = function()
            return os.date('%Y-%m-%d', os.time() - 86400)
          end,
          today = function()
            return os.date '%Y-%m-%d'
          end,
          tomorrow = function()
            return os.date('%Y-%m-%d', os.time() + 86400)
          end,

          day = function()
            return os.date '%d'
          end,
          month = function()
            return os.date '%m'
          end,
          week = function()
            return os.date '%W'
          end,
          weekday = function()
            return os.date('*t').wday
          end,
          weekdayname = function()
            local daysoftheweek =
              { 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday' }
            return daysoftheweek[os.date('*t').wday]
          end,
          year = function()
            return os.date '%Y'
          end,
        },
      },
    },
  }
```

### Environment

```
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1692716794

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/__w/neovim/neovim/build/nvim.AppDir/usr/share/nvim"

Run :checkhealth for more info
Obsidian.nvim v3.7.3 (0f4ea504f51e3d1591e90184967c75503106a89b)
Status:
   buffer directory: nil
   working directory: /home/gbbirkisson/repos/personal/notes
Workspaces:
  ✓ active workspace: Workspace(name='personal', path='/home/gbbirkisson/repos/personal/notes', root='/home/gbbirkisson/repos/personal/notes')
Dependencies:
   plenary.nvim: f7adfc4b3f4f91aab6caebf42b3682945fbc35be
Integrations:
   picker: TelescopePicker()
   completion: enabled (nvim-cmp)
Tools:
   rg: ripgrep 14.1.0
Environment:
   operating system: Linux
Config:
   notes_subdir: nil
```

**Comment by zDonik1** [comment]
Somewhat related to the current bug: if the `disable_frontmatter` is set to `true`, then the frontmatter from the daily template is removed entirely. From the README:

> ```lua
>   -- Optional, boolean or a function that takes a filename and returns a boolean.
>   -- `true` indicates that you don't want obsidian.nvim to manage frontmatter.
>   disable_frontmatter = false,
> ```

Disabling frontmatter doesn't mean removing it, but making sure that it is not edited by obsidian.nvim.

@epwalsh I have already been looking through the template creation code, and am open to collaborate on the issue if this is considered a bug.

**Comment by epwalsh** [comment]
@zDonik1 yes I would consider this is a bug. PR welcome!

**Comment by epwalsh** [comment]
@gbbirkisson fixed in 4e80ab4.

**Comment by gbbirkisson** [comment]
Thanks a lot. Loving this plugin :heart: Keep up the amazing work!
